src/synthcity/plugins/core/models/tabular_ddpm/utils.py

```diff
@@ -0,0 +1,230 @@
+# mypy: disable-error-code=no-untyped-def
+
+# stdlib
+from inspect import isfunction
+
+# third party
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+# from torch.profiler import record_function
+
+
+def normal_kl(mean1, logvar1, mean2, logvar2):
+    """
+    Compute the KL divergence between two gaussians.
+
+    Shapes are automatically broadcasted, so batches can be compared to
+    scalars, among other use cases.
+    """
+    tensor = None
+    for obj in (mean1, logvar1, mean2, logvar2):
+        if isinstance(obj, torch.Tensor):
+            tensor = obj
+            break
+    assert tensor is not None, "at least one argument must be a Tensor"
```

Comment:
Having asserts in public code it is not safe. Replace this with exceptions instead

Reply:
I have replace all the asserts.